.github/workflows/build-image.yaml

```diff
@@ -6,19 +6,25 @@ jobs:
   prepare:
     runs-on: ubuntu-latest
     outputs:
-      FULL_IMAGE_TAG: ${{ steps.prefix.outputs.prefix }}${{ steps.short-sha.outputs.sha }}
+      FULL_IMAGE_TAG: ${{ steps.tag.outputs.tag }}
     steps:
-      - name: Get Short SHA
-        uses: benjlevesque/short-sha@v1.2
-        id: short-sha
-      - name: Set Prefix
-        id: prefix
+      - name: Set Tag
+        id: tag
         run: |
-          if [ ${GITHUB_REF##*/} = "main" ]; then
-            echo "::set-output name=prefix::stg-"
+          export CI_COMMIT_SHORT_SHA=$(echo $GITHUB_SHA | cut -c1-8)
+          echo $CI_COMMIT_SHORT_SHA;
+          echo ${GITHUB_REF##*/};
+          echo $version_pattern;
+          if [[ ${GITHUB_REF##*/} =~ $version_pattern ]]; then
```

Comment:
I think this will also push branches named ``v0.0.0`` to production.

* Branch `refs/heads/v0.0.0`
* Tag `refs/tags/v0.0.0`
* PR `refs/pull/121/merge`

I think we also need to match on the `/tags/` part, not just the last component.